tests/rule_based_profiler/domain_builder/test_domain_builder.py

```diff
@@ -0,0 +1,184 @@
+from great_expectations.validator.validator import Validator
+
+
+# noinspection PyPep8Naming
+def test_active_batch_table_domain_builder(
+    pandas_test_df,
+    table_Users_domain,
+):
+    batch_definition: BatchDefinition = BatchDefinition(
+        datasource_name="my_datasource",
+        data_connector_name="my_data_connector",
+        data_asset_name="my_data_asset",
+        batch_identifiers=IDDict({}),
+    )
+
+    validator: Validator = build_pandas_validator_with_data(
+        df=pandas_test_df,
+        batch_definition=batch_definition,
+    )
+
+    domain_builder: DomainBuilder = ActiveBatchTableDomainBuilder()
+    domains: List[Domain] = domain_builder.get_domains(
+        validator=validator,
+    )
+
+    assert len(domains) == 1
+    assert domains == [
+        {
+            "domain_type": "table",
+            "domain_kwargs": {
+                "batch_id": "f576df3a81c34925978336d530453bc4",
+            },
+        }
+    ]
+
+    domain: Domain = domains[0]
+    # Assert Domain object equivalence.
+    assert domain == table_Users_domain
+    # Also test that the dot notation is supported properly throughout the dictionary fields of the Domain object.
+    assert domain.domain_kwargs.batch_id == "f576df3a81c34925978336d530453bc4"
+
+
+# noinspection PyPep8Naming
+def test_column_domain_builder(
+    pandas_test_df,
+    column_Age_domain,
+    column_Date_domain,
+    column_Description_domain,
+):
+    batch_definition: BatchDefinition = BatchDefinition(
+        datasource_name="my_datasource",
+        data_connector_name="my_data_connector",
+        data_asset_name="my_data_asset",
+        batch_identifiers=IDDict({}),
+    )
+
+    validator: Validator = build_pandas_validator_with_data(
+        df=pandas_test_df,
+        batch_definition=batch_definition,
+    )
+
+    domain_builder: DomainBuilder = ColumnDomainBuilder()
+    domains: List[Domain] = domain_builder.get_domains(
+        validator=validator,
+    )
+
+    assert len(domains) == 3
+    assert domains == [
+        {
+            "domain_type": "column",
+            "domain_kwargs": {
+                "column": "Age",
+                "batch_id": "f576df3a81c34925978336d530453bc4",
+            },
+        },
+        {
+            "domain_type": "column",
+            "domain_kwargs": {
+                "column": "Date",
+                "batch_id": "f576df3a81c34925978336d530453bc4",
+            },
+        },
+        {
+            "domain_type": "column",
+            "domain_kwargs": {
+                "column": "Description",
+                "batch_id": "f576df3a81c34925978336d530453bc4",
+            },
+        },
+    ]
+    # Assert Domain object equivalence.
+    domain: Domain
+    domain = domains[0]
+    assert domain == column_Age_domain
+    domain = domains[1]
+    assert domain == column_Date_domain
+    domain = domains[2]
+    assert domain == column_Description_domain
```

Comment:
Instead of reassigning `domain` to list items of `domains` each time, could we just e.g. `assert domain[0] == column_Age_domain`?

Reply:
@talagluck we could -- doing this in steps is a matter of style (e.g., for clarity).